app/src/main/java/org/wordpress/pioupiou/postlist/PostListFragment.java

```diff
@@ -53,21 +58,20 @@ public void onCreate(Bundle savedInstanceState) {
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.fragment_post_list, container, false);
-
-        // Set the adapter
-        if (view instanceof RecyclerView) {
-            Context context = view.getContext();
-            RecyclerView recyclerView = (RecyclerView) view;
-            if (mColumnCount <= 1) {
-                recyclerView.setLayoutManager(new LinearLayoutManager(context));
-            } else {
-                recyclerView.setLayoutManager(new GridLayoutManager(context, mColumnCount));
-            }
-            recyclerView.setAdapter(new PostRecyclerViewAdapter(DummyContent.ITEMS, mListener));
+        mRecyclerView = (RecyclerView) view.findViewById(R.id.list);
+        if (mColumnCount <= 1) {
+            mRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
+        } else {
+            mRecyclerView.setLayoutManager(new GridLayoutManager(getActivity(), mColumnCount));
         }
+
         return view;
     }
 
+    void setPosts(@NonNull AccountModel account, @NonNull List<PostModel> posts) {
+        mRecyclerView.setAdapter(new PostRecyclerViewAdapter(getActivity(), account, posts, mListener));
```

Comment:
Wouldn't it be better to add a method like `update(List<PostModel>)` to the `PostRecyclerViewAdapter` that we call here on updates after initial setup (rather than instantiating a new `PostRecyclerViewAdapter` each time)?